### PR TITLE
C++ API parity: ReLU6

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27538 C++ API parity: Threshold
* #27537 C++ API parity: Tanhshrink
* #27536 C++ API parity: Tanh
* #27535 C++ API parity: Softsign
* #27534 C++ API parity: Softshrink
* #27489 C++ API parity: Softplus
* #27488 C++ API parity: Sigmoid
* #27487 C++ API parity: CELU
* #27437 C++ API parity: RReLU
* **#27436 C++ API parity: ReLU6**

Differential Revision: [D17835414](https://our.internmc.facebook.com/intern/diff/D17835414)

### CODE REVIEW

File: test/cpp/api/modules.cpp [hunk 1]
```diff
@@ -1422,3 +1439,9 @@ TEST_F(ModulesTest, PrettyPrintReLU) {
   ASSERT_EQ(c10::str(ReLU(ReLUOptions().inplace(true))),
     "torch::nn::ReLU(inplace=true)");
 }
+
+TEST_F(ModulesTest, PrettyPrintReLU6) {
+  ASSERT_EQ(c10::str(ReLU6()), "torch::nn::ReLU6()");
+  ASSERT_EQ(c10::str(ReLU6(ReLU6Options().inplace(true))),
+    "torch::nn::ReLU6(inplace=true)");
+}
```
Comment (yf225) [inline test/cpp/api/modules.cpp]: nit: need new line here.

File: torch/csrc/api/src/nn/modules/activation.cpp [hunk 2]
```diff
@@ -134,5 +134,23 @@ void ReLUImpl::pretty_print(std::ostream& stream) const {
...
+Tensor ReLU6Impl::forward(Tensor& input) {
+  return F::relu6(input, options);
+}
+
+void ReLU6Impl::reset() {}
+
+void ReLU6Impl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::ReLU6(";
+  if (options.inplace()) {
+    stream << std::boolalpha  << "inplace=" << options.inplace();
```
Comment (yf225) [inline torch/csrc/api/src/nn/modules/activation.cpp]: nit:
```suggestion
    stream << std::boolalpha << "inplace=" << options.inplace();
```

File: torch/csrc/api/include/torch/nn/modules/activation.h [hunk 3]
```diff
@@ -169,5 +169,28 @@ class TORCH_API ReLUImpl : public torch::nn::Cloneable<ReLUImpl> {
...
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ReLU6 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// Applies the ReLU6 function element-wise.
+/// See https://pytorch.org/docs/master/nn.html#torch.nn.ReLU6 to learn
+/// about the exact behavior of this module.
+class TORCH_API ReLU6Impl : public torch::nn::Cloneable<ReLU6Impl> {
+ public:
+  ReLU6Impl() : ReLU6Impl(ReLU6Options()) {}
+  explicit ReLU6Impl(const ReLU6Options& options_);
```
Comment (yf225) [inline torch/csrc/api/include/torch/nn/modules/activation.h]: similarly here: would be awesome to combine these two constructors.

File: torch/csrc/api/include/torch/nn/functional/activation.h [hunk 4]
```diff
@@ -52,6 +52,11 @@ inline Tensor relu(Tensor& input, const ReLUOptions& options) {
   }
 }
 
+inline Tensor relu6(Tensor& input, const ReLU6Options& options) {
```
Comment (yf225) [inline torch/csrc/api/include/torch/nn/functional/activation.h]: similarly here: `options` can be optional, and we should add a test for this too.

File: test/cpp/api/functional.cpp [hunk 5]
```diff
@@ -409,3 +409,32 @@ TEST_F(FunctionalTest, ReLUDefaultOptions) {
...
   ASSERT_TRUE(torch::allclose(y, y_exp));
 }
+
+TEST_F(FunctionalTest, ReLU6) {
+  const auto size = 3;
+  for (const auto inplace : {false, true}) {
+    auto x = torch::linspace(-10.0, 10.0, size * size * size);
+    x.resize_({size, size, size});
+    auto y_exp = (x < 0) * 0 + ((x >= 0) * (x <= 6)) * x + (x > 6) * 6;
+    auto y = F::relu6(x, ReLU6Options().inplace(inplace));
```
Comment (yf225) [inline test/cpp/api/functional.cpp]: similarly here: would be great to test `F::relu6(x, /*inplace=*/inplace)`.

File: test/cpp/api/modules.cpp [hunk 6]
```diff
@@ -1443,3 +1460,9 @@ TEST_F(ModulesTest, PrettyPrintReLU) {
   ASSERT_EQ(c10::str(ReLU(ReLUOptions().inplace(true))),
     "torch::nn::ReLU(inplace=true)");
 }
+
+TEST_F(ModulesTest, PrettyPrintReLU6) {
+  ASSERT_EQ(c10::str(ReLU6()), "torch::nn::ReLU6()");
+  ASSERT_EQ(c10::str(ReLU6(ReLU6Options().inplace(true))),
+    "torch::nn::ReLU6(inplace=true)");
```
Comment (yf225) [inline test/cpp/api/modules.cpp]: similarly here: would be great to test `ReLU6(/*inplace=*/true)`.

**Review by yf225 — APPROVED** [review]
